src/bootstrap/src/core/build_steps/perf.rs

```diff
@@ -1,12 +1,115 @@
+use std::fmt::{Display, Formatter};
 use std::process::Command;
 
 use crate::core::build_steps::compile::{Std, Sysroot};
 use crate::core::build_steps::tool::RustcPerf;
 use crate::core::builder::Builder;
 use crate::core::config::DebuginfoLevel;
 
+/// Performs profiling or benchmarking with [`rustc-perf`](https://github.com/rust-lang/rustc-perf)
+/// using a locally built compiler.
+#[derive(Debug, Clone, clap::Parser)]
+pub struct PerfArgs {
+    #[clap(subcommand)]
+    cmd: PerfCommand,
+
+    #[clap(flatten)]
+    opts: SharedOpts,
+}
+
+impl Default for PerfArgs {
+    fn default() -> Self {
+        Self { cmd: PerfCommand::Eprintln, opts: SharedOpts::default() }
+    }
+}
+
+#[derive(Debug, Clone, clap::Parser)]
+enum PerfCommand {
+    /// Run `profile_local eprintln`.
+    /// This executes the compiler on the given benchmarks and stores its stderr output.
+    Eprintln,
+    /// Run `profile_local samply`
+    /// This executes the compiler on the given benchmarks and profiles it with `samply`.
+    /// You need to install `samply`, e.g. using `cargo install samply`.
```

Comment:
Without `--locked` flag the installation may fail.

```suggestion
    /// You need to install `samply`, e.g. using `cargo install --locked samply`.
```

Reply:
Good catch, I actually encountered this myself when installing `samply` to test this :laughing: